app/models/case/sar/offender.rb

```diff
@@ -83,9 +83,8 @@ def searchable_fields_and_ranks
   before_validation :reassign_gov_uk_dates
   before_save :set_subject
 
-  def set_subject
-    self.subject = subject_full_name
-  end
+  before_save :use_subject_as_requester,
+              if: -> { name.blank? }
 
```

Comment:
Assuming the `if: -> ...` on the new line is type?

In terms of function naming, should this just be called `set_subject_as_case_name` (I subscribe to the ronseal school of function naming)

In terms of callback, should this be a `before_validation` rather than `before_save`? Would that make any desirable/undesirable difference e.g. on the front-end, for other validation rules or code that depends on having a `name` before the details are all saved?

Reply:
All of this is exactly how it is in sar/standard.rb